			     BASH PATCH REPORT
			     =================

Bash-Release:	5.1
Patch-ID:	bash51-007

Bug-Reported-by:	Tom Tromey <tom@tromey.com>
Bug-Reference-ID:	<875z3u9fd0.fsf@tromey.com>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-readline/2021-01/msg00009.html

Bug-Description:

The code to check readline versions in an inputrc file had the sense of the
comparisons reversed.

--- a/lib/readline/bind.c
+++ b/lib/readline/bind.c
@@ -1234,7 +1234,7 @@ parser_if (char *args)
 #endif /* VI_MODE */
   else if (_rl_strnicmp (args, "version", 7) == 0)
     {
-      int rlversion, versionarg, op, previ, major, minor;
+      int rlversion, versionarg, op, previ, major, minor, opresult;
 
       _rl_parsing_conditionalized_out = 1;
       rlversion = RL_VERSION_MAJOR*10 + RL_VERSION_MINOR;
@@ -1294,24 +1294,25 @@ parser_if (char *args)
       switch (op)
 	{
 	case OP_EQ:
-	  _rl_parsing_conditionalized_out = rlversion == versionarg;
+ 	  opresult = rlversion == versionarg;
 	  break;
 	case OP_NE:
-	  _rl_parsing_conditionalized_out = rlversion != versionarg;
+	  opresult = rlversion != versionarg;
 	  break;
 	case OP_GT:
-	  _rl_parsing_conditionalized_out = rlversion > versionarg;
+	  opresult = rlversion > versionarg;
 	  break;
 	case OP_GE:
-	  _rl_parsing_conditionalized_out = rlversion >= versionarg;
+	  opresult = rlversion >= versionarg;
 	  break;
 	case OP_LT:
-	  _rl_parsing_conditionalized_out = rlversion < versionarg;
+	  opresult = rlversion < versionarg;
 	  break;
 	case OP_LE:
-	  _rl_parsing_conditionalized_out = rlversion <= versionarg;
+	  opresult = rlversion <= versionarg;
 	  break;
 	}
+      _rl_parsing_conditionalized_out = 1 - opresult;
     }
   /* Check to see if the first word in ARGS is the same as the
      value stored in rl_readline_name. */
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 6
+#define PATCHLEVEL 7
 
 #endif /* _PATCHLEVEL_H_ */
